ucp/public_api.py

```diff
@@ -298,27 +389,56 @@ def closed(self):
         nbytes: int, optional
             Number of bytes to receive. Default is the whole buffer.
         """
-        await self._ep.recv(buffer, nbytes=nbytes)
+        if self._closed:
+            raise exceptions.UCXCloseError("Endpoint closed")
+        nbytes = buffer_interface.get_buffer_nbytes(
+            buffer, check_min_size=nbytes, cuda_support=self._cuda_support
+        )
+        log = "[Recv #%03d] ep: %s, tag: %s, nbytes: %d" % (
+            self._recv_count,
+            hex(self.uid),
+            hex(self._msg_tag_recv),
+            nbytes,
+        )
+        logging.debug(log)
+        self.pending_msg_list.append({"log": log})
+        self._recv_count += 1
+        tag = self._msg_tag_recv
+        if self._guarantee_msg_order:
+            tag += self._recv_count
+
+        ret = await send_recv.tag_recv(
+            self._worker, buffer, nbytes, tag, pending_msg=self.pending_msg_list[-1]
+        )
+        self._finished_recv_count += 1
+        if (
+            self._close_after_n_recv is not None
+            and self._finished_recv_count >= self._close_after_n_recv
+        ):
+            self.abort()
+        return ret
 
     def ucx_info(self):
         """Return low-level UCX info about this endpoint as a string"""
-        return self._ep.ucx_info()
+        if self._closed:
+            raise exceptions.UCXCloseError("Endpoint closed")
+        return self._ep.info()
 
     def cuda_support(self):
         """Return whether UCX is configured with CUDA support or not"""
-        return self._ep._cuda_support
+        return self._cuda_support
 
     def get_ucp_worker(self):
         """Returns the underlying UCP worker handle (ucp_worker_h)
         as a Python integer.
         """
-        return self._ep.get_ucp_worker()
+        return self._worker.handle
 
     def get_ucp_endpoint(self):
         """Returns the underlying UCP endpoint handle (ucp_ep_h)
         as a Python integer.
         """
-        return self._ep.get_ucp_endpoint()
+        return self._ep.handle
```

Comment:
Is this valid if the endpoint is closed or should there be a check/exception before?

Reply:
You are right, I am adding some close checks

Reply:
Thanks for adding those.